control-plane/connect-inject/endpoints_controller_ent_test.go

```diff
@@ -1639,61 +1954,56 @@ func TestReconcileDeleteEndpointWithNamespaces(t *testing.T) {
 					// Create a token for it if ACLs are enabled.
 					if tt.enableACLs {
```

Comment:
What do you think about simplifying it the inside of this `if` condition like so:

```
var writeOpts api.WriteOptions
if svc.Kind == api.ServiceKindMeshGateway {
	writeOpts.Namespace = "default"
} else {
	writeOpts.Namespace = ts.ConsulNS
}
writeOpts.Namespace = "default" // Mesh Gateways must always be registered in the "default" namespace.
test.SetupK8sAuthMethod(t, consulClient, svc.Service, svc.Meta[MetaKeyKubeNS])
token, _, err = consulClient.ACL().Login(&api.ACLLoginParams{
	AuthMethod:  test.AuthMethod,
	BearerToken: test.ServiceAccountJWTToken,
	Meta: map[string]string{
		TokenMetaPodNameKey: fmt.Sprintf("%s/%s", svc.Meta[MetaKeyKubeNS], svc.Meta[MetaKeyPodName]),
		"component":         svc.ID,
	},
}, &writeOpts)

require.NoError(t, err)
```

It looks like the namespace in `WriteOpts` is the only thing different between the cases.

Reply:
Great call!